src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,51 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array;
+    private int size;
 
+    public MateStack() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    private void grow() {
+        Object[] newArray = new Object[array.length + (array.length >> 1)];
+        System.arraycopy(array, 0, newArray, 0, array.length);
+        array = (T[]) newArray;
+    }
+
+    public void push(T value) {
+        if (size == array.length) {
+            grow();
+        }
+        array[size] = value;
+        size++;
     }
 
     public T peek() {
-        return null;
+        checkException();
+        return array[size - 1];
     }
 
     public T pop() {
-        return null;
+        checkException();
+        T element = array[size - 1];
+        System.arraycopy(array, size + 1, array, size - 1, 0);
+        size--;
```

Comment:
how many elements do you copy on line 37?

let's just make the last element null instead

Reply:
done